### PR TITLE
/policycompass/policycompass#279 (Visualisation datasets with unit_id=0)

### PR DESCRIPTION
**Pull request by mmilaprat**
this PR solves the visualisation bug that appears when a dataset has a unit_id=0.

/policycompass/policycompass/issues/279

### CODE REVIEW

**Comment by th0br0** [comment]
__ref__ policycompass/policycompass#279

**Comment by mmilaprat** [comment]
Who must accept this PR?

File: app/modules/visualization/controllers/visualizationControllers.js [hunk 1]
```diff
@@ -2430,6 +2434,12 @@ angular.module('pcApp.visualization.controllers.visualization', [
 					//console.log("-------------->"+arguments[i]['unit_id']);
 					
 					$dataUnit[i] = Unit.getById(arguments[i]['unit_id']);
```
Comment (cbotsikas) [inline app/modules/visualization/controllers/visualizationControllers.js]: Why did you keep the request for unit info even in the case of 0? It produces an error "GET https://services-stage.policycompass.eu/api/v1/references/units/0 404 (NOT FOUND)" at the console. Shouldn't line 2436 (including the promise at line 2444) be inside the else case of the check you added at line 2437?

**Comment by cbotsikas** [comment]
This commit solves the infinite loop with the indicator charts but i guess we should wait for @mmilaprat to address my inline comment

**Comment by mmilaprat** [comment]
Now we avoid to recover unit details when unit id=0.

**Pull request merged**